### PR TITLE
Document inline arrays and more unit tests

### LINKED ISSUE #984: Document inline arrays
### Description

We call them 'ad-hoc arrays' in the docs. ODFF uses name inline arrays and we should be consistent with it. We need to describe their usage and limitations discussed in #982 

### Links

#982

### Type of issue

- [x] Missing documentation
- [ ] Error in the documentation
- [ ] Request for a demo
- [ ] Request for a tutorial, guide etc.
- [ ] Other issues, suggestions or ideas

### PR DESCRIPTION
**Pull request by sequba**
### Context

Improve the documentation of inline arrays according to #982 and #984 

### How has this been tested?

build works, unit tests pass

### Types of changes

- [x] Documentation change

### Related issue(s):
1. Fixes #984 
2. #982 
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [ ] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: docs/guide/arrays.md [hunk 1]
```diff
@@ -6,14 +6,28 @@ Use array formulas to perform an operation (or call a function) on multiple cell
...
 
 An array is inherently a two-dimensional object.
 
 `1`x`1` arrays are treated as single, zero-dimensional values (**scalars**).
 
+### Inline arrays
+
+Inline arrays are defined by curly braces. They might contain one or more rows of data separated by `arrayColumnSeparator` (default: `,`) and `arrayRowSeparator` (default: `;`). All the rows of an inline array must be of equal length.
+
+**Important: An inline array is never recomputed after initialization. If it contains a cell reference, an inline array will not be updated when the value of the cell changes.**
```
Comment (sequba) [inline docs/guide/arrays.md:21]: @kirszenbaum Any tips on better wordings?

File: docs/guide/known-limitations.md [hunk 2]
```diff
@@ -54,6 +54,7 @@ The inconsistencies are due to:
...
+| Cell references inside inline arrays               | ={A1, A2}                                                                 | Array value calculated but not updated on cell values changes.                                                                                                                                                                                                                                         | Array value calculated and updated on cell values changes.                                                                        | ERROR - invalid array                                    |
```
Comment (kirszenbaum) [inline docs/guide/known-limitations.md:57]:
```suggestion
| Cell references inside inline arrays               | ={A1, A2}                                                                 | The array's value is calculated but not updated when the cells' values change.                                                                                                                                                                                                                                         | The array's value is calculated and updated when the cells' values change.                                                                        | ERROR: invalid array                                    |
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

File: docs/guide/arrays.md (hunk 1 above)
Comment (kirszenbaum) [reply docs/guide/arrays.md:21]: @sequba - I added my ideas

File: docs/guide/arrays.md [hunk 3]
```diff
@@ -6,14 +6,28 @@ Use array formulas to perform an operation (or call a function) on multiple cell
...
+
+```js
+= {1, 2, 3} // single row
+= {1, 2 ; 3, 4} // two rows
+= {1, 2 ; 3 } // INVALID: rows of different length
+= SUM({1, 2, 3}) // inline array as a function argument
+= {A1, A2} // will not be updated when A1 or A2 changes
+```
```
[the remark this answers is not in the document]
Comment (sequba) [reply docs/guide/arrays.md:29]: Great! Thanks a lot @kirszenbaum

File: docs/guide/arrays.md (hunk 1 above)
Comment (sequba) [reply docs/guide/arrays.md]: Thanks a lot @kirszenbaum

File: docs/guide/known-limitations.md (hunk 2 above)
Comment (sequba) [reply docs/guide/known-limitations.md]: Suggestion applied

File: docs/guide/arrays.md [hunk 4]
```diff
@@ -6,14 +6,28 @@ Use array formulas to perform an operation (or call a function) on multiple cell
 
 In HyperFormula, an array can be:
 * A range of cell addresses (e.g. `A1:A10`)
-* An **ad-hoc array**: an input array with no ground-truth range that defines the values (e.g. `{1,3,5}`)
+* An **inline array**: an ad-hoc array not referring to any range of cells (e.g. `{1,3,5}`)
```
[the remark this answers is not in the document]
Comment (sequba) [reply docs/guide/arrays.md]: Suggestions applied

**Review by kirszenbaum — APPROVED** [review]

**Pull request merged**